docs/articles_en/learn-openvino/large-language-models/llm-inference-native-ov.rst

```diff
@@ -0,0 +1,167 @@
+
+Use the model and tokenizer converted from the previous step:
+
+.. code-block:: python
+
+  import numpy as np
+  from openvino import compile_model
+  from openvino_tokenizers import unpack_strings
+
+  # Compile the tokenizer, model, and detokenizer using OpenVINO. These files are XML representations of the models optimized for OpenVINO
+  compiled_tokenizer = compile_model("openvino_tokenizer.xml")
+  compiled_model = compile_model("openvino_model.xml")
+  compiled_detokenizer = compile_model("openvino_detokenizer.xml")
+
+2.	Tokenize and Transform Input
++++++++++++++++++++++++++++++++++++++++
+
+Tokenization is a mandatory step in the process of generating text using LLMs. Tokenization
+converts the input text into a sequence of tokens, which are essentially the format that the
+model can understand and process. The input text string must be tokenized and set up in the
+structure expected by the model before running inference.
+
+.. code-block:: python
+
+  text_input = ["Quick brown fox was"]
+  ov_input = compiled_tokenizer(text_input)
+
+3.	Generate Tokens
++++++++++++++++++++++++++++++++++++++++
+
+The core of text generation lies in the inference and token selection loop. In each iteration
+of this loop, the model runs inference on the input sequence, generates and selects a new token,
+and appends it to the existing sequence.
+
+.. code-block:: python
+
+  # Define the number of new tokens to generate
+  new_tokens_size = 10
+
+  # Determine the size of the existing prompt
+  prompt_size = ov_input["input_ids"].shape[-1]
+
+  # Prepare the input dictionary for the model
+  # It combines existing tokens with additional space for new tokens
+  input_dict = {
+    output.any_name: np.hstack([tensor, np.zeros(shape=(1, new_tokens_size), dtype=np.int_)])
+    for output, tensor in ov_input.items()
+  }
+
+  # Generate new tokens iteratively
+  for idx in range(prompt_size, prompt_size + new_tokens_size):
+      # Get output from the model
+      output = compiled_model(input_dict)["token_ids"]
+      # Update the input_ids with newly generated token
+      input_dict["input_ids"][:, idx] = output[:, idx - 1]
+      # Update the attention mask to include the new token
+      input_dict["attention_mask"][:, idx] = 1
+
+4.	Decode and Display Output
++++++++++++++++++++++++++++++++++++++++
+
+The final step in the process is de-tokenization, where the sequence of token IDs generated by
+the model is converted back into human-readable text.
+This step is essential for interpreting the model's output.
+
+.. code-block:: python
+
+  # Extract token IDs for the final output
+  ov_token_ids = input_dict["input_ids"]
+  # Decode the model output back to string
+  ov_output = unpack_strings(compiled_detokenizer(ov_token_ids)["string_output"])
+  print(f"OpenVINO output string: `{ov_output}`")
+
+.. code-block:: python
+
+  # Example output:
+  ['<s> Quick brown fox was walking through the forest. He was looking for something']
```

Comment:
In the end, I would recommend adding "Additional Resources" section with the link to gen.ai repo, namely [here](https://github.com/openvinotoolkit/openvino.genai/tree/master/text_generation/causal_lm/cpp)